File: java/resistor-color/src/main/java/ResistorColor.java
class ResistorColor {

    private static String[] _colors =
            { "black", "brown", "red", "orange", "yellow", "green", "blue",
                    "violet", "grey", "white"};

    int colorCode(String color) {
        // Would prefer to use a switch or a map but the tests seem to lean
        // towards using a String array.
        return java.util.Arrays.asList(_colors).indexOf(color);
    }

    String[] colors() {
        return _colors;
    }
}

File: c/resistor-color/src/resistor_color.c
#include "resistor_color.h"

int color_code(resistor_band_t c)
{
	return c;
}

resistor_band_t* colors()
{
	static resistor_band_t all_colors[] = {
	  BLACK, BROWN, RED, ORANGE, YELLOW,
	  GREEN, BLUE, VIOLET, GREY, WHITE
	};

	return all_colors;
}
